code/game/objects/structures/crates_lockers/closets/secure/personal.dm

```diff
@@ -9,11 +9,22 @@
 	if(registered_name)
 		to_chat(user, "<span class='notice'>The display reads, \"Owned by [registered_name]\".</span>")
 
-/obj/structure/closet/secure_closet/personal/check_access(obj/item/card/id/I)
+/obj/structure/closet/secure_closet/personal/check_access(obj/item/I)
 	. = ..()
 	if(!I || !istype(I))
 		return
-	if(registered_name == I.registered_name)
+	if(istype(I,/obj/item/modular_computer/tablet))
+		var/obj/item/modular_computer/tablet/ourTablet = I
+		var/obj/item/computer_hardware/card_slot/card_slot = ourTablet.all_components[MC_CARD]
+		if(card_slot)
+			return registered_name == card_slot.stored_card.registered_name || registered_name == card_slot.stored_card2.registered_name
+	var/obj/item/card/id/ID
+	if(istype(I,/obj/item/pda))
+		var/obj/item/pda/ourPDA = I
+		ID = ourPDA.id
+	else if(istype(I,/obj/item/card/id))
+		ID = I
+	if(registered_name == ID.registered_name)
 		return TRUE

```

Comment:
Just do the following, dude.
```
	var/obj/item/card/id/ID = I.getID()
	if(!ID || registered_name != ID.registered_name)
		return FALSE
	return TRUE
```

Reply:
doesn't actually work for tablets as such, since it only checks the one ID for that, but yeah, apparently so